homeassistant/components/loqed/lock.py

```diff
@@ -0,0 +1,269 @@
+        wh_id = Undefined
+        # Check if hook already registered @loqed
+        for hook in webhooks:
+            if hook["url"].startswith(
+                self._internal_url + "/api/webhook/" + WEBHOOK_PREFIX
+            ):
+                url = hook["url"]
+                wh_id = WEBHOOK_PREFIX + url[-12:]
+                _LOGGER.debug("Found already configured webhook @loqed: %s", url)
+                break
+        if wh_id == Undefined:
+            wh_id = WEBHOOK_PREFIX + get_random_string(12)
+            # Registering webhook in Loqed
+            url = self._internal_url + "/api/webhook/" + wh_id
+            _LOGGER.debug("Registering webhook @loqed: %s", url)
+            await self._lock.registerWebhook(url)
+        # Registering webhook in HASS, when exists same will be used
+        _LOGGER.debug("Registering webhook in HA")
+        self._webhook = str(url)
+        try:
+            webhook.async_register(
+                hass=self.hass,
+                domain=DOMAIN,
+                name="loqed",
+                webhook_id=wh_id,
+                handler=self.async_handle_webhook,
+            )
+        except ValueError:  # when already installed
+            pass
+        return url
+
+    @callback
+    async def async_handle_webhook(self, hass, webhook_id, request):
+        """Handle webhook callback."""
+        _LOGGER.debug("Callback received: %s", str(request.headers))
+        received_ts = request.headers["TIMESTAMP"]
+        received_hash = request.headers["HASH"]
+        body = await request.text()
+        _LOGGER.debug("Callback body: %s", body)
+        event_data = await self._lock.receiveWebhook(body, received_hash, received_ts)
+        if "error" in event_data:
+            _LOGGER.warning("Incorrect CALLBACK RECEIVED:: %s", event_data)
+            return
+        event_type = "LOQED_status_change_to_" + LOCK_STATES[self.bolt_state]
+        _LOGGER.debug("Firing event:: %s", event_type)
+        hass.bus.fire(event_type, event_data)
+        self.async_schedule_update_ha_state(False)
+        event = event_data["event_type"].strip().lower()
+        if event.split("_")[0] == "state":
+            if self.update_task:
+                self.update_task.cancel()
+        elif "go_to" in event:
+            await self.async_schedule_update(12)
+
+    async def async_schedule_update(self, timeout):
+        """To cancel outstanding async update task and schedules new one."""
+        if self.update_task:
+            self.update_task.cancel()
+        _LOGGER.debug("PLAN update operation in %s seconds", timeout)
+        self.update_task = asyncio.create_task(self.async_delayed_update(timeout))
+
+    async def async_delayed_update(self, timeout):
+        """Async update task to handle lock update when nno callback."""
+        _LOGGER.debug("Start waiting in delayed_update")
+        await asyncio.sleep(timeout)
+        await self.async_update()
```

Comment:
You might want to consider using our `DataUpdateCoordinator` instead. It can cooldown and schedule update requests. It would also centralize your data handling for this integration nicely (making it easier to add more platforms like sensors and switch in the future).